lib/cocoapods/installer/analyzer.rb

```diff
@@ -554,11 +563,20 @@ def filter_pod_targets_for_target_definition(target_definition, pod_targets_by_t
             whitelists = dependencies.map do |dependency|
               target_definition.pod_whitelisted_for_configuration?(dependency.name, configuration_name)
             end.uniq
+            if resolver_spec.transitive?
```

Comment:
instead of introducing a whole new property, could you check `target_dependency.dependencies.any? {|d| d.root_name == resolver_spec.name }` or somethign like that?

Reply:
we could use `resolver_spec.spec.dependencies` and whitelist any names included there if the direct dependency itself gets whitelisted, but that would require multiple iterations over the list of pod targets to make sure we don't bail early if the whitelisted dependent comes after the one which is not whitelisted. I think that would increase the complexity of this implementation